plasma_framework/contracts/src/vaults/Erc20Vault.sol

```diff
@@ -0,0 +1,50 @@
+pragma solidity ^0.5.0;
+pragma experimental ABIEncoderV2;
+
+import "./Vault.sol";
+import "./predicates/IErc20DepositVerifier.sol";
+import {PaymentTransactionModel as DepositTx} from "../transactions/PaymentTransactionModel.sol";
+import "openzeppelin-solidity/contracts/token/ERC20/IERC20.sol";
+import "openzeppelin-solidity/contracts/token/ERC20/SafeERC20.sol";
+
+contract Erc20Vault is Vault {
+    IErc20DepositVerifier private _depositVerifier;
+
+    using SafeERC20 for IERC20;
+
+    constructor(address _blockController) Vault(_blockController) public {}
+
+    /**
+     * @notice Set the deposit verifier contract. This can be only called by the operator.
+     * @param _contract address of the verifier contract.
+     */
+    function setDepositVerifier(address _contract) public onlyOperator {
+        _depositVerifier = IErc20DepositVerifier(_contract);
+    }
+
+    /**
+     * @notice Deposits approved amount of ERC20 token. Approve must have been called first.
+     * @param _depositTx RLP encoded transaction to act as the deposit.
+     */
+    function deposit(bytes calldata _depositTx) external {
+        DepositTx.Transaction memory decodedTx = DepositTx.decode(_depositTx);
+
+        _depositVerifier.verify(decodedTx, msg.sender);
+
+        IERC20 erc20 = IERC20(decodedTx.outputs[0].token);
```

Comment:
As mentioned in another comment, we cannot hardcode our tx model dependency here if we want to have the benefit to upgrade to new deposit tx type. In the case here, I guess to be able to do so we have to return some value from the "verifier" interface.

eg. (pseudo code)
```
(owner, token, amount) = depositVerifier.verify(_depositTx, msg.sender); 
erc20 = IERC20(token);
erc20.safeTransferFrom(msg.sender, address(this), amount);
```

btw, I feel 
```
require(erc20.allowance(msg.sender, address(this)) == decodedTx.outputs[0].amount, "Tokens have not been approved");
```
can be done inside verifier?